### PR TITLE
VMware Prow/Test-Grid Updates

### PR DESCRIPTION
**Pull request by akutz**
This patch introduces several updates to the VMware jobs and test-grid reports:

Pre-submits:
* pull-cloud-provider-vsphere-check - provides linting
* pull-cloud-provider-vsphere-build - builds the CCM and CSI binaries
* pull-cloud-provider-vsphere-unit-test - executes the unit tests
* pull-cloud-provider-vsphere-integration-test - executes the integration tests

Post-submits:
* post-cloud-provider-vsphere-deploy - builds, tests, and deploys the CCM and CSI images

Periodics for out-of-tree cloud provider:
* ci-cloud-provider-vsphere-conformance-latest
* ci-cloud-provider-vsphere-conformance-stable-1-13
* ci-cloud-provider-vsphere-conformance-stable-1-12
* ci-cloud-provider-vsphere-conformance-stable-1-11

Periodics for in-tree cloud provider:
* ci-vsphere-conformance-latest
* ci-vsphere-conformance-stable-1-13
* ci-vsphere-conformance-stable-1-12
* ci-vsphere-conformance-stable-1-11

cc @figo @frapposelli

### CODE REVIEW

**Comment by akutz** [comment]
Related to https://github.com/kubernetes/cloud-provider-vsphere/pull/166

**Comment by akutz** [comment]
Blocked by https://github.com/kubernetes/cloud-provider-vsphere/pull/167

**Comment by akutz** [comment]
Hi @neolit123,

I'd really appreciate it if you could take a look at this and make any suggestions or provide any feedback you might have. Thank you in advance!

File: config/jobs/kubernetes/cloud-provider-vsphere/cloud-provider-vsphere-config.yaml [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+    - image: gcr.io/cloud-provider-vsphere/ci:a74fdada
+      command:
+      - "make"
+      args:
+      - "conformance-test"
+    env:
+    - name: CLOUD_PROVIDER
+      value: vsphere
+    - name: K8S_VERSION
+      value: ci/latest.txt
```
Comment (neolit123) [inline config/jobs/kubernetes/cloud-provider-vsphere/cloud-provider-vsphere-config.yaml]: should this be `release/stable-1.13.txt`?

Comment (akutz) [reply config/jobs/kubernetes/cloud-provider-vsphere/cloud-provider-vsphere-config.yaml]: Hi @neolit123,

It should. I had in fact caught it when updating for the failed bazel job. Should be good now. Thanks!

File: config/jobs/kubernetes/cloud-provider-vsphere/cloud-provider-vsphere-config.yaml [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+      - image: gcr.io/cloud-provider-vsphere/ci:a74fdada
+        command:
+        - "/bin/bash"
+        args:
+        - "-c"
+        - "make unit-test && make integration-test && make push-images"
+      securityContext:
+        privileged: true
+
+perodics:
```
Comment (neolit123) [inline config/jobs/kubernetes/cloud-provider-vsphere/cloud-provider-vsphere-config.yaml]: `periodics`

File: config/jobs/kubernetes/cloud-provider-vsphere/cloud-provider-vsphere-config.yaml [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+        args:
+        - "integration-test"
+      securityContext:
+        privileged: true
+
+postsubmits:
+  kubernetes/cloud-provider-vpshere:
+
+  # Deploys the CCM and CSI images if both the unit and integration tests succeed.
+  - name: post-cloud-provider-vsphere-deploy
```
Comment (neolit123) [inline config/jobs/kubernetes/cloud-provider-vsphere/cloud-provider-vsphere-config.yaml]: unused in testgrid, it seems.

**Comment by akutz** [comment]
/hold cancel

Okay @neolit123, this should pass the tests. I finally got them running locally so I didn't have to keep submitting just to find the errors.

**Comment by neolit123** [comment]
> Okay @neolit123, this should pass the tests. I finally got them running locally so I didn't have to keep submitting just to find the errors.

the github status is missing in this PR for some reason, but LGTM.

> I would make this change except there's no stable 1.14 yet:

ok, i guess the 1.11 tests can be kept until the 1.14 stable.

**Comment by akutz** [comment]
Hi @figo,

Can you please lgtm this again? The alert addresses have been updated.

**Comment by spiffxp** [comment]
@akutz 
>/hold

What's the condition for removing the /hold ?

**Comment by akutz** [comment]
Hi @spiffxp,

I just wanted to triple check it since it's been a minute since I submitted it.

**Pull request merged**